crates/controller/src/hud/minimap/interaction.rs

```diff
@@ -63,68 +78,137 @@ impl MinimapClickEvent {
     }
 }
 
-impl fmt::Debug for MinimapClickEvent {
+impl fmt::Debug for MinimapPressEvent {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(f, "{:?} -> {:?}", self.button, self.position)
     }
 }
 
-fn click_handler(
+#[derive(Event)]
+struct MinimapDragEvent {
+    button: MouseButton,
+    position: Vec2,
+}
+
+impl MinimapDragEvent {
+    fn new(button: MouseButton, position: Vec2) -> Self {
+        Self { button, position }
+    }
+
+    fn button(&self) -> MouseButton {
+        self.button
+    }
+
+    /// Position on the map in 2D flat coordinates (these are not minimap
+    /// coordinates).
+    fn position(&self) -> Vec2 {
+        self.position
+    }
+}
+
+#[derive(Resource, Deref, DerefMut)]
+struct DraggingButtons(Vec<MouseButton>);
+
+fn press_handler(
     window_query: Query<&Window, With<PrimaryWindow>>,
     mut input_events: EventReader<MouseButtonInput>,
     hud: HudNodes<With<MinimapNode>>,
     bounds: Res<MapBounds>,
-    mut click_events: EventWriter<MinimapClickEvent>,
+    mut dragging: ResMut<DraggingButtons>,
+    mut press_events: EventWriter<MinimapPressEvent>,
 ) {
-    let Some(cursor) = window_query.single().cursor_position() else {
-        return;
-    };
+    let cursor = window_query.single().cursor_position();
+
     for event in input_events.iter() {
-        if event.state != ButtonState::Released {
+        if event.state != ButtonState::Pressed {
```

Comment:
I would `match event.state` so that it doesn't break when other `ButtonState`s are added in the future.